cosmos/operators/airflow_async.py

```diff
@@ -35,8 +43,91 @@ class DbtSourceAirflowAsyncOperator(DbtSourceLocalOperator):
+        self.full_refresh = full_refresh
+
+        # airflow task param
+        self.configuration: dict[str, object] = {}
+        self.gcp_conn_id = self.profile_config.profile_mapping.conn_id  # type: ignore
+        if not self.profile_config or not self.profile_config.profile_mapping:
+            raise CosmosValueError(f"Cosmos async support is only available starting in Airflow 2.8 or later.")
+        profile = self.profile_config.profile_mapping.profile
+        self.gcp_project = profile["project"]
+        self.dataset = profile["dataset"]
+        self.location = kwargs.get("location", "northamerica-northeast1")  # TODO: must be provided by users
+        super().__init__(
+            *args, configuration=self.configuration, location=self.location, task_id=kwargs["task_id"], deferrable=True
+        )
+
+        if self.profile_type not in _SUPPORTED_DATABASES:
+            raise CosmosValueError(f"Async run are only supported: {_SUPPORTED_DATABASES}")
+
+    def get_remote_sql(self) -> str:
+        if not settings.AIRFLOW_IO_AVAILABLE:
+            raise CosmosValueError(f"Cosmos async support is only available starting in Airflow 2.8 or later.")
+        from airflow.io.path import ObjectStoragePath
+
+        file_path = self.extra_context["dbt_node_config"]["file_path"]
+        dbt_dag_task_group_identifier = self.extra_context["dbt_dag_task_group_identifier"]
+
+        remote_target_path_str = str(remote_target_path).rstrip("/")
+        project_dir_parent = str(self.project_dir.parent)
+        relative_file_path = str(file_path).replace(project_dir_parent, "").lstrip("/")
+        remote_model_path = f"{remote_target_path_str}/{dbt_dag_task_group_identifier}/compiled/{relative_file_path}"
+
+        print("remote_model_path: ", remote_model_path)
+        object_storage_path = ObjectStoragePath(remote_model_path, conn_id=remote_target_path_conn_id)
+        with object_storage_path.open() as fp:  # type: ignore
+            return fp.read()  # type: ignore
+
+    def drop_table_sql(self) -> None:
+        model_name = self.task_id.split(".")[0]
+        sql = f"DROP TABLE IF EXISTS {self.gcp_project}.{self.dataset}.{model_name};"
+        hook = BigQueryHook(
+            gcp_conn_id=self.gcp_conn_id,
+            impersonation_chain=self.impersonation_chain,
+        )
+        self.configuration = {
+            "query": {
+                "query": sql,
+                "useLegacySql": False,
+            }
+        }
+        hook.insert_job(configuration=self.configuration, location=self.location, project_id=self.gcp_project)
+
+    def execute(self, context: Context) -> Any | None:
+        if not self.full_refresh:
+            raise CosmosValueError("The async execution only supported for full_refresh")
+        else:
+            self.drop_table_sql()
+
+            sql = self.get_remote_sql()
+            model_name = self.task_id.split(".")[0]
```

Comment:
This fails when we define `DbtTaskGroup` instead of `DbtDag` as the tasks have ID as e.g. `task_group.stg_customers.run` when converted from `DbtTaskGroup` instead of `stg_customers.run` as observed when converted from `DbtDag`

Reply:
As a result it creates such a table and keeps overwriting that table for each of the tasks in the task group. dbt tests fail eventually as it cannot find the needed tables like stg_customers, stg_orders, etc.
<img width="255" alt="Screenshot 2024-10-02 at 12 22 53 AM" src="https://github.com/user-attachments/assets/05bfeb46-d058-40b0-927b-ace3fba1ae8e">

Reply:
Pushed a fix for this in https://github.com/astronomer/astronomer-cosmos/pull/1230/commits/72c6164480c553b12c366bc11fc9d9e049515100